shared/nm-default.h

```diff
@@ -22,19 +22,95 @@
 #ifndef __NM_DEFAULT_H__
 #define __NM_DEFAULT_H__
 
-/* makefiles define NETWORKMANAGER_COMPILATION for compiling NetworkManager.
- * Depending on which parts are compiled, different values are set. */
-#define NM_NETWORKMANAGER_COMPILATION_DEFAULT             0x0001
-#define NM_NETWORKMANAGER_COMPILATION_INSIDE_DAEMON       0x0002
-#define NM_NETWORKMANAGER_COMPILATION_LIB                 0x0004
-#define NM_NETWORKMANAGER_COMPILATION_SYSTEMD             0x0008
-#define NM_NETWORKMANAGER_COMPILATION_LIB_LEGACY          0x0010
+#define NM_NETWORKMANAGER_COMPILATION_WITH_GLIB                 (1 <<  0)
```

Comment:
It seems to me that 

```
NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_PRIVATE
NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_CORE_INTERNAL
NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_CORE_PRIVATE
NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_UTIL_PRIVATE
NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_GLIB_PRIVATE
NM_NETWORKMANAGER_COMPILATION_WITH_SYSTEMD
```

are not used anywhere; can they be dropped?

Reply:
`NM_NETWORKMANAGER_COMPILATION_WITH_SYSTEMD` is actually needed. That was a bug.

Regarding the others: maybe. The `*_PRIVATE` flag indicates that you are allowed to make use of private API -- that is, you are building the library itself. Although unused, I added them for completeness, hoping that the code shows the intend of how it should/could be used (instead of some random comment that gets ignored). To me, it should be a very different thing whether you build libnm-core.a or libnm.so. The former can access `libnm-core/nm-utils-private.h`, the latter not. Since those private headers are in the same path as the public headers, we only don't include them by convention -- which is easy to get wrong. One idea was to do:
```
    #if !((NETWORKMANAGER_COMPILATION) & NM_NETWORKMANAGER_COMPILATION_WITH_LIBNM_PRIVATE)
    #error Cannot use private header
    #endif
```
in those headers.

But maybe such headers should instead move to separate directories, like
```
   libnm-core/private/nm-utils-private.h
   libnm-core/internal/nm-core-internal.h
```
and when building libnm-core.a, you need to explicitly specify those include paths. Downside is, that our list of `-I` arguments is already long, so I am not convinced that more directories are better.


Ok, I now dropped the libnm-util and libnm-glib releated define, and made use of all others. If course, if you don't agree that they are used in a useful manner, you could argue that it's an invalid excuse for having them :)

Reply:
> Ok, I now dropped the libnm-util and libnm-glib releated define, and made use of all others. If course, if you don't agree that they are used in a useful manner, you could argue that it's an invalid excuse for having them :)

Looks good.